src/hera/dag.py

```diff
@@ -122,15 +121,20 @@ def __exit__(self, exc_type, exc_val, exc_tb) -> None:
 
     def add_task(self, t: Task) -> "DAG":
         """Add a task to the DAG. Note that tasks are added automatically when the DAG context is used"""
-        add_task(self, t)
+        if t is None:
+            return self
```

Comment:
`t` is not marked `Optional`, so this should never happen. Or, does the hint need to be `: Optional[Task]`? I don't think one should be allowed to pass in a `None`.

Similar in `add_tasks` (also, could that one just call `self.add_task` inside the loop?)